streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStateManagerImpl.java

```diff
@@ -334,10 +335,33 @@ public void close(final Map<TopicPartition, Long> offsets) throws IOException {
 
     @Override
     public void checkpoint(final Map<TopicPartition, Long> offsets) {
+
+        // Find non persistent store's topics
+        Map<String, String> storeToChangelogTopic = topology.storeToChangelogTopic();
+        Set<String> globalNonPersistentStoresTopics = new HashSet<>();
+        for (StateStore store : topology.globalStateStores()) {
+            if (!store.persistent() && storeToChangelogTopic.containsKey(store.name())) {
+                globalNonPersistentStoresTopics.add(storeToChangelogTopic.get(store.name()));
+            }
+        }
+
         checkpointableOffsets.putAll(offsets);
-        if (!checkpointableOffsets.isEmpty()) {
+
+        final Map<TopicPartition, Long> filteredOffsets = new HashMap<>();
```

Comment:
Do we need `filteredOffsets` or can we just filter `offsets` directly?

Reply:
`checkpointableOffsets` is set on other places as well. Filtering `offsets` only unfortunately is not enough. 

I did not want to cause a trouble so I tried to protect further unexpected bugs as explained above. This seems to best way to protect it.

Reply:
Please see my other comment: for non-persistent stores the corresponding `checkpointableOffsets` entry should be `NO_CHECKPOINT`.